src/lib/datepicker/datepicker.ts

```diff
@@ -138,6 +139,13 @@ export class MatDatepickerContent<D> extends _MatDatepickerContentMixinBase
     });
   }
 
+  _handleKeydown(event: KeyboardEvent) {
+    // Closing on alt + up is only valid when there's an input associated with the datepicker.
+    if (this.datepicker._datepickerInput && event.altKey && event.keyCode === UP_ARROW) {
```

Comment:
The <kbd>Esc</kbd> appears to be handled on the `overlayRef` (line 475), should this be the same?

Reply:
Totally forgot that we had that. Changed.